Docker file

KristapsEisaks03 opened the pull request:
I think this is self explanatory, feel free to ask me questions if you got any

KristapsEisaks03 committed:
Just testing

KristapsEisaks03 committed:
TESTINGGG

KristapsEisaks03 committed:
Actual working - single file dockerfile

KristapsEisaks03 commented:
Need to create docker compose file and have a look more into it

KristapsEisaks03 committed:
Created docker-compose file

KristapsEisaks03 commented:
Added docker compose file and tested!

KristapsEisaks03 committed:
Fixes

- now includes only single RUN
- now uses a different web server implementation

KristapsEisaks03 commented:
The RUN command has been changed including the deployment (however still wouldn't call it ready for the actual deployment)

KristapsEisaks03 committed:
pre-commits

- tested pre commit and added ruff

KristapsEisaks03 committed:
.

KristapsEisaks03 committed:
Merge pull request #5 from andmatei/pre-commits

Pre commits

KristapsEisaks03 committed:
Merge branch 'main' into EXPERIMENT-Docker

andmatei commented on Dockerfile:
```diff
@@ -0,0 +1,23 @@
+
+# Expose port for the application
+EXPOSE 8000
+
```
[non-blocking] I think you should also have a line here that runs the migrations - it would be a good idea to have an `entrypoint.sh` script that handles all that. This can be added in a following PR when you make to change to PostgreSQL.

andmatei reviewed (dismissed):
Great job! Just left a comment to keep in mind for a future PR

KristapsEisaks03 committed:
Final docker update

- added entrypoint.sh file for docker
- had to change the dockerfile itself as --dev option was depracated

andmatei commented on entrypoint.sh:
```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
+
+# Activate the virtual environment
+. /app/.venv/bin/activate
```
I think you should let poetry do this

andmatei commented on entrypoint.sh:
```diff
@@ -0,0 +1,11 @@
+. /app/.venv/bin/activate
+
+# Run database migrations
+python manage.py makemigrations
```
No need to run this, it should be run by anyone making changes to the models

andmatei commented on entrypoint.sh:
```diff
@@ -0,0 +1,11 @@
+
+# Run database migrations
+python manage.py makemigrations
+python manage.py migrate
```
`poetry run python manage.py migrate`
